storm-client/src/jvm/org/apache/storm/trident/spout/RichSpoutBatchExecutor.java

```diff
@@ -194,7 +194,12 @@ public void reportError(Throwable t) {
         public void emitDirect(int task, String stream, List<Object> values, Object id) {
             throw new UnsupportedOperationException("Trident does not support direct streams");
         }
-        
+
+        @Override
+        public void flush() {
+            //NOOP   //TODO: Roshan: validate if this is OK
```

Comment:
TODO needs to go away.  Is this OK to not have a flush?

Reply:
needs to flush.